UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingAbility/ChangelingBaseAbility.cs

```diff
@@ -0,0 +1,95 @@
+using Changeling;
+using Mirror;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Changeling
+{
+	public class ChangelingBaseAbility: ActionData, ICooldown
+	{
+		public short Index => (short)ChangelingAbilityList.Instance.Abilites.IndexOf(this);
+
+		[SerializeField] protected int cooldown = 1;
+		public float DefaultTime => cooldown;
+
+		[Tooltip("Evolution points cost for buying")]
+		[SerializeField] protected int abilityEPCost;
+		public int AbilityEPCost => abilityEPCost;
+		[Tooltip("Chemical points cost for use")]
+		[SerializeField] protected int abilityChemCost;
+		public int AbilityChemCost => abilityChemCost;
+
+		[SerializeField] protected GameObject abilityImplementation = null;
+		public GameObject AbilityImplementation => abilityImplementation;
+		public override bool CallOnClient => true;
+		public override bool CallOnServer => false;
+
+		[TextArea(8, 20)]
+		[Tooltip("Description that will be used in store")]
+		[SerializeField] protected string descriptionStore = "";
+		[SerializeField] private bool canBeUsedWhileInCrit = false;
+		public bool CanBeUsedWhileInCrit => canBeUsedWhileInCrit;
+		[SerializeField] private bool isLocal = false;
+		public bool IsLocal => isLocal;
+		public string DescriptionStore
+		{
+			get
+			{
+				if (descriptionStore == null || descriptionStore.Length == 0)
+					return description;
+				return descriptionStore;
+			}
+		}
+
+		[Tooltip("Shows only when absorbed someone")]
+		[SerializeField] private bool showsOnlyWhenAbsorbedSomeone = false;
+		public bool ShowsOnlyWhenAbsorbedSomeone => showsOnlyWhenAbsorbedSomeone;
+
+		[Tooltip("Is ability will be added on start")]
+		public bool startAbility = false;
```

Comment:
what ability is available from start but doesn't require to have absorbed someone? Maybe this field is redundant.

Reply:
Gonna remove because not used